src/pip/_internal/network/session.py

```diff
@@ -76,136 +70,275 @@
 ]
 
 
-# These are environment variables present when running under various
-# CI systems.  For each variable, some CI systems that use the variable
-# are indicated.  The collection was chosen so that for each of a number
-# of popular systems, at least one of the environment variables is used.
-# This list is used to provide some indication of and lower bound for
-# CI traffic to PyPI.  Thus, it is okay if the list is not comprehensive.
-# For more background, see: https://github.com/pypa/pip/issues/5499
-CI_ENVIRONMENT_VARIABLES = (
-    # Azure Pipelines
-    "BUILD_BUILDID",
-    # Jenkins
-    "BUILD_ID",
-    # AppVeyor, CircleCI, Codeship, Gitlab CI, Shippable, Travis CI
-    "CI",
-    # Explicit environment variable.
-    "PIP_IS_CI",
-)
+class Telemetry:
+    """Return a string representing the user agent.
+
+    FIXME: This string is currently propagated as a header into every single HTTP
+           request pip makes.  It should really be subject to a formal PEP process
+           in order to ensure user consent around telemetry is respected.
```

Comment:
Please propose this in https://discuss.python.org/c/packaging/14, not in the pip doc strings. It's not clear to me that everyone will agree with this.